### PR TITLE
Remove scm stage from pr pipeline

### PR DESCRIPTION
**Pull request by pablochacin**
## Why is this PR needed?

Presently the PR code is rebased on top of the target branch as part of the pipeline logic. This has several issues: 
* The PR code is checked (linted) before rebasing to the target branch (e.g. master)
* The pipeline code that is executed is the one from the PR, not the more recent from the target branch, if any.
* The logic for checking-out PR and merging with target branch is split between the Jenkins pipeline and python code in `ci/tasks/rebase.py` executed from the `ci/Makefile`. This makes it difficult to maintain this logic avoiding conflicts and inconsistencies.

Fixes https://github.com/SUSE/avant-garde/issues/1698

## What does this PR do?

Removes the logic for rebasing the code from the PR pipeline and uses the marge option in the job's scm configuration. This way, the PR is merged before the pipeline is executed.

Due to a limitation in the multibranch job plugin, this change impacts the location of the code, as it cannot be checked out into a subdirectory as the current logic does, but directly into the workspace. Therefore, the daily jobs must also be updated to checkout the code directly in the workspace and the code location must be adjusted in all scripts and pipelines.

Also, because the code is merged before the pipeline is executed, the reference to the head of the PR branch is lost and the logic for reporting the status to github must be adapted.

Finally, the unit test make target had to be fixed to support the checkout into a directory different from `skuba`.

## Anything else a reviewer needs to know?

The PR has been tested with the new SCM configuration in [this modified pr pipeline](https://ci.suse.de/job/caasp-jobs/job/dev/job/pablo/job/pr-test-pablo/view/change-requests/job/PR-1168/)

Also, a [daily e2e job has been tested against this PR](https://ci.suse.de/job/caasp-jobs/job/e2e/job/v5/job/vmware/job/test_nginx_deployment-daily/30/) with the modified SCM configuration (checkout into the workspace, not into the skuba subdirectory)

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

File: ci/Makefile [hunk 1]
```diff
@@ -55,10 +55,6 @@ cleanup:
 gather_logs:
 	${TEST_RUNNER} --platform ${PLATFORM} get_logs
 
-.PHONY: git_rebase_check
-git_rebase_check:
-	${TASKS}/${REBASE}
```
Comment (JanZerebecki) [inline ci/Makefile:60]: Then `REBASE := rebase.py` above can also be deleted.

Comment (pablochacin) [reply ci/Makefile:60]: Good catch, I'll do in a follow-up PR just not to have to ask again for aprovals.

Comment (pablochacin) [reply ci/Makefile:60]: Good catch. Thanks

File: ci/jenkins/pipelines/prs/skuba-update-acceptance.Jenkinsfile [hunk 2]
```diff
@@ -51,7 +51,7 @@ pipeline {
         } } }
 
         stage('Setting GitHub in-progress status') { steps {
-            sh(script: "${PR_MANAGER} update-pr-status ${GIT_COMMIT} ${PR_CONTEXT} 'pending'", label: "Sending pending status")
+            sh(script: "${PR_MANAGER} update-pr-status ${PR_CONTEXT} 'pending'", label: "Sending pending status")
         } }
 
         stage('Git Clone') { steps {
```
Comment (JanZerebecki) [inline ci/jenkins/pipelines/prs/skuba-update-acceptance.Jenkinsfile:57]: Probably remove?

File: ci/jenkins/pipelines/prs/skuba-update-unit.Jenkinsfile [hunk 3]
```diff
@@ -52,7 +52,7 @@ pipeline {
         } } }
 
         stage('Setting GitHub in-progress status') { steps {
-            sh(script: "${PR_MANAGER} update-pr-status ${GIT_COMMIT} ${PR_CONTEXT} 'pending'", label: "Sending pending status")
+            sh(script: "${PR_MANAGER} update-pr-status ${PR_CONTEXT} 'pending'", label: "Sending pending status")
         } }
 
         stage('Git Clone') { steps {
```
Comment (JanZerebecki) [inline ci/jenkins/pipelines/prs/skuba-update-unit.Jenkinsfile:58]: Probably remove?

File: ci/jenkins/pipelines/prs/skuba-jjb-validation.Jenkinsfile [hunk 4]
```diff
@@ -51,7 +51,7 @@ pipeline {
         } } }
 
         stage('Setting GitHub in-progress status') { steps {
-            sh(script: "${PR_MANAGER} update-pr-status ${GIT_COMMIT} ${PR_CONTEXT} 'pending'", label: "Sending pending status")
+            sh(script: "${PR_MANAGER} update-pr-status ${PR_CONTEXT} 'pending'", label: "Sending pending status")
         } }
 
         stage('Git Clone') { steps {
```
Comment (JanZerebecki) [inline ci/jenkins/pipelines/prs/skuba-jjb-validation.Jenkinsfile:57]: Probably remove?

Comment (pablochacin) [reply ci/jenkins/pipelines/prs/skuba-jjb-validation.Jenkinsfile:57]: Not sure what you are asking here. Remove what?

Comment (JanZerebecki) [reply ci/jenkins/pipelines/prs/skuba-jjb-validation.Jenkinsfile:57]: The stage that reclones the repo under the subdirectory that this PR removes.

File: ci/jenkins/pipelines/prs/skuba-update-acceptance.Jenkinsfile (hunk 2 above)
Comment (pablochacin) [reply ci/jenkins/pipelines/prs/skuba-update-acceptance.Jenkinsfile:57]: Not related to this PR, will do in a follow up PR.

Comment (JanZerebecki) [reply ci/jenkins/pipelines/prs/skuba-update-acceptance.Jenkinsfile:57]: Thought that was the reason for the CI failure... ok, if its not.

**Comment by JanZerebecki** [comment]
Compared to my last review this is a rebase without any changes. CI fails like before. I still think removing the mentioned git clone steps would fix it.

**Comment by pablochacin** [comment]
> Compared to my last review this is a rebase without any changes. CI fails like before. I still think removing the mentioned git clone steps would fix it.

Yes, you were right. Thanks for insisting!

**Comment by pablochacin** [comment]
Update acceptance is failing, but it is because the test is broken in v5.

**Pull request merged**